tracing-appender/src/lib.rs

```diff
@@ -7,6 +7,19 @@ pub mod non_blocking;
 pub mod rolling;
 mod worker;
 
+/// A function for which accepts a struct implementing the `Write` Trait and provides a
+/// a `MakeWriter` which allows writes to happen in a non blocking manner.
+/// # Examples
+/// ``` rust,ignore
+/// use tracing_subscriber::fmt::MakeWriter;
+///
+/// let (non_blocking, _guard) = non_blocking(std::io::stdout());
```

Comment:
also, note that these examples are compiled _outside_ of this crate (as though they were user code that depends on it), so you need to either import `non_blocking` here or use the full path to it:
```suggestion
/// let (non_blocking, _guard) = tracing_appender::non_blocking(std::io::stdout());
```
personally, I think it's better to use the prefixed path, because it makes it clearer to users looking at this that `non_blocking` is the function from _this_ crate.